src/pickup.cpp

```diff
@@ -470,16 +471,18 @@ bool pick_one_up( const tripoint &pickup_target, item &newit, vehicle *veh,
             picked_up = u.wear_item( newit );
             break;
         case WIELD:
-            picked_up = u.wield( newit );
-            if( !picked_up ) {
-                break;
+            if ( wield_check.success() ) {
+                u.wield( newit );
```

Comment:
`picked_up` is not set on WIELD action, so subsequent code is not called:

```c++
    if( picked_up ) {
        remove_from_map_or_vehicle( pickup_target, veh, cargo_part, moves_taken, index );
    }
```

Reply:
Oh, will revert and resend.